checkbox-support/checkbox_support/scripts/run_watcher.py

```diff
@@ -93,124 +90,171 @@ def run(self):
-        detected after ACTION_TIMEOUT secs
+        Reserve or removal the detected storage info.
+
+        write the info we got in this script to $PLAINBOX_SESSION_SHARE
+        so the other jobs, e.g. read/write test, could know more information,
+        for example the partition it want to try to mount.
+
+        :param reserve:
+            type: Boolean
+            - True: backup the info of storage partition to PLAINBOX_SESSION_SHARE
+            - False: remove the backup file from PLAINBOX_SESSION_SHARE
+        :param storage_type:
+            type: String
+            - Type of storage. e.g. usb, mediacard and thunderbolt
+        :param mounted_partition:
+            type: String
+            - The name of partition. e.g. sda1, nvme1n1p1 etc...
         """
-        logger.error(
-            "no {} storage {} was reported in systemd journal".format(
-                self.args.storage_type, self.args.testcase
+        plainbox_session_share = os.environ.get("PLAINBOX_SESSION_SHARE")
+        if not plainbox_session_share:
+            logger.error("no env var PLAINBOX_SESSION_SHARE")
+            sys.exit(1)
+
+        # TODO: Should name the file by the value of storage_type variable as
+        #       prefix. e.g. thunderbolt_insert_info, mediacard_insert_info.
+        #       Since usb_insert_info is used by usb_read_write script, we
+        #       should refactor usb_read_write script to adopt different files
+        file_name = "usb_insert_info"
+
+        # backup the storage partition info
+        if reserve and mounted_partition:
+            logger.info(
+                "cache file {} is at: {}".format(
+                    file_name, plainbox_session_share
+                )
             )
-        )
-        sys.exit(1)
+            file_to_share = open(
+                os.path.join(plainbox_session_share, file_name), "w"
+            )
+            file_to_share.write(mounted_partition + "\n")
+            file_to_share.close()
+
+        # remove the back info
+        if not reserve:
+            file_to_share = os.path.join(plainbox_session_share, file_name)
+            with contextlib.suppress(FileNotFoundError):
+                os.remove(file_to_share)
 
 
-class USBStorage(StorageInterface):
+class USBStorage(StorageWatcher):
     """
-    USBStorage hanldes the insertion and removal of usb2, usb3 and mediacard.
+    USBStorage handles the insertion and removal of usb2, usb3 and mediacard.
     """
 
-    MOUNTED_PARTITION = None
-    FLAG_DETECTION = {
-        "device": {
-            "new high-speed USB device number": False,
-            "new SuperSpeed USB device number": False,
-            "new SuperSpeed Gen 1 USB device number": False,
-        },
-        "driver": {"using ehci_hcd": False, "using xhci_hcd": False},
-        "insertion": {"USB Mass Storage device detected": False},
-        "removal": {"USB disconnect, device number": False},
-    }
+    class Device(Enum):
```

Comment:
Please add a docstring to this and consider changing the name to... `DeviceName` or something similar

Minor: you can use `class DeviceName(str, Enum)` so that below you can simply say
```
for device_name in DeviceName:
  if device_name in line: ...
```

Reply:
ditto for the other lines, in general it is not clear what these are when you use them in the code